### PR TITLE
Update Broker flight API automation test

### PR DESCRIPTION
**Pull request by rpdome**
Update Broker Flight API to cover getFlights(), setFlights()
also added steps in https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1561087

### CODE REVIEW

**Review by fadidurah — APPROVED** [review]
Looks good, thanks dome!

**Review by shahzaibj — APPROVED** [review]

**Review by somalaya — APPROVED** [review]

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java [hunk 1]
```diff
@@ -44,8 +44,28 @@ public void test_1561087() {
...
+        mBroker.addFlights(anotherFlightJson);
+        Assert.assertEquals( "{\"AnotherFlight\":\"hello\",\"SetFlightsTest\":\"true\"}", mBroker.getFlights());
+
+        // Override flights and get to check if the flight information is returned
+        final String flightToOverwrite = "{\"SetFlightsTest\":\"false\"}";
+        mBroker.addFlights(flightToOverwrite);
+        Assert.assertEquals("{\"AnotherFlight\":\"hello\",\"SetFlightsTest\":\"false\"}", mBroker.getFlights());
+
+        // Add flight with null value. the flight map should not change.
+        final String flightMapWithNullValue = "{\"SetFlightsTest\": null}";
```
Comment (iamgusain) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java:58]: what does setting null means?
should it remove the flight/reset to default value?
if we want to no-op it, should we return an error/ throw exception?

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java [hunk 2]
```diff
@@ -44,8 +44,28 @@ public void test_1561087() {
...
+        mBroker.addFlights(flightMapWithNullValue);
+        Assert.assertEquals("{\"AnotherFlight\":\"hello\",\"SetFlightsTest\":\"false\"}", mBroker.getFlights());
+
+        // Add an empty flight map. the flight map should not change.
+        final String emptyFlightMap = "{}";
+        mBroker.addFlights(emptyFlightMap);
```
Comment (iamgusain) [inline msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java]: Should this throw IllegalOperationException?

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java (hunk 1 above)
Comment (rpdome) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java:58]: modified. Setting null means we'll remove the flag - if it exists.

File: msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java (hunk 2 above)
Comment (rpdome) [reply msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/brokerapi/TestCase1561087.java]: This is a legal input (not a malformed json).

**Pull request merged**